--- esp-idf/examples/openthread/ot_rcp/main/esp_ot_rcp.c	2026-04-02 18:00:13.377299900 -0700
+++ examples/openthread/ot_rcp/main/esp_ot_rcp.c	2026-04-02 18:57:17.007812200 -0700
@@ -21,6 +21,11 @@
 #include "esp_ot_config.h"
 #include "esp_vfs_eventfd.h"
 
+// required for xiao_select_external_antenna()
+#include "driver/gpio.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+
 #if CONFIG_ESP_COEX_EXTERNAL_COEXIST_ENABLE
 #include "ot_examples_common.h"
 #endif
@@ -37,8 +42,17 @@
 
 extern void otAppNcpInit(otInstance *instance);
 
-void app_main(void)
-{
+// external antenna patch for XIAO ESP32-C6
+static void xiao_select_external_antenna(void) {
+    gpio_set_direction(GPIO_NUM_3, GPIO_MODE_OUTPUT);
+    gpio_set_level(GPIO_NUM_3, 0);
+    vTaskDelay(pdMS_TO_TICKS(100));
+    gpio_set_direction(GPIO_NUM_14, GPIO_MODE_OUTPUT);
+    gpio_set_level(GPIO_NUM_14, 1);
+}
+
+void app_main(void) {
+    xiao_select_external_antenna();
     // Used eventfds:
     // * ot task queue
     // * radio driver
